File: palestra-transform-otim-obj-kotlin/src/main/java/transformacao/java/dtoscomplexos/objetos/dto/VendaDTO.java
package transformacao.java.dtoscomplexos.objetos.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public class VendaDTO {
    private LocalDateTime dataVenda;
    private ClienteDTO cliente;
    private BigDecimal valorTotal;
    private List<VendaItemDTO> itensVenda;

    public LocalDateTime getDataVenda() {
        return dataVenda;
    }

    public void setDataVenda(LocalDateTime dataVenda) {
        this.dataVenda = dataVenda;
    }

    public ClienteDTO getCliente() {
        return cliente;
    }

    public void setCliente(ClienteDTO cliente) {
        this.cliente = cliente;
    }

    public BigDecimal getValorTotal() {
        return valorTotal;
    }

    public void setValorTotal(BigDecimal valorTotal) {
        this.valorTotal = valorTotal;
    }

    public List<VendaItemDTO> getItensVenda() {
        return itensVenda;
    }

    public void setItensVenda(List<VendaItemDTO> itensVenda) {
        this.itensVenda = itensVenda;
    }
}

File: palestra-transform-otim-obj-kotlin/src/main/kotlin/transformacao/kotlin/dtoscomplexos/beans/VendaDTO.kt
package transformacao.kotlin.dtoscomplexos.beans

import java.math.BigDecimal
import java.time.LocalDateTime

data class VendaDTO(
        val numero: Int,
        val cliente: String,
        val endereco: String,
        val valorTotal: BigDecimal,
        val dataVenda: LocalDateTime,
        val itens: List<VendaItemDTO>
)

data class VendaItemDTO(
        val quantidade: BigDecimal,
        val valorUnitario: BigDecimal,
        val valorTotal: BigDecimal,
        val produto: ProdutoDTO
)

data class ProdutoDTO(
        val nome: String,
        val valorUnitario:BigDecimal
)